cli/lsp/tsc.rs

```diff
@@ -0,0 +1,1061 @@
+use regex::Captures;
+use regex::Regex;
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+fn display_parts_to_string(
+  maybe_parts: Option<Vec<SymbolDisplayPart>>,
+) -> Option<String> {
+  if let Some(parts) = maybe_parts {
+    Some(
+      parts
+        .into_iter()
+        .map(|p| p.text)
+        .collect::<Vec<String>>()
+        .join(""),
+    )
+  } else {
+    None
+  }
+}
+
+fn get_tag_body_text(tag: &JSDocTagInfo) -> Option<String> {
+  if let Some(text) = &tag.text {
+    match tag.name.as_str() {
+      "example" => {
+        let caption_regex =
+          Regex::new(r"<caption>(.*?)</caption>\s*\r?\n((?:\s|\S)*)").unwrap();
+        if caption_regex.is_match(text) {
+          Some(
+            caption_regex
+              .replace(text, |c: &Captures| {
+                format!("{}\n\n{}", &c[1], make_codeblock(&c[2]))
+              })
+              .to_string(),
+          )
+        } else {
+          Some(make_codeblock(text))
+        }
+      }
+      "author" => {
+        let email_match_regex =
+          Regex::new(r"(.+)\s<([-.\w]+@[-.\w]+)>").unwrap();
+        Some(
+          email_match_regex
+            .replace(text, |c: &Captures| format!("{} {}", &c[1], &c[2]))
+            .to_string(),
+        )
+      }
+      "default" => Some(make_codeblock(text)),
+      _ => Some(replace_links(text)),
+    }
+  } else {
+    None
+  }
+}
+
+fn get_tag_documentation(tag: &JSDocTagInfo) -> String {
+  match tag.name.as_str() {
+    "augments" | "extends" | "param" | "template" => {
+      if let Some(text) = &tag.text {
+        let part_regex = Regex::new(r"^(\S+)\s*-?\s*").unwrap();
+        let body: Vec<&str> = part_regex.split(&text).collect();
+        if body.len() == 3 {
+          let param = body[1];
+          let doc = body[2];
+          let label = format!("*@{}* `{}`", tag.name, param);
+          if doc.is_empty() {
+            return label;
+          }
+          if doc.contains('\n') {
+            return format!("{}  \n{}", label, replace_links(doc));
+          } else {
+            return format!("{} - {}", label, replace_links(doc));
+          }
+        }
+      }
+    }
+    _ => (),
+  }
+  let label = format!("*@{}*", tag.name);
+  let maybe_text = get_tag_body_text(tag);
+  if let Some(text) = maybe_text {
+    if text.contains('\n') {
+      format!("{}  \n{}", label, text)
+    } else {
+      format!("{} - {}", label, text)
+    }
+  } else {
+    label
+  }
+}
+
+fn make_codeblock(text: &str) -> String {
+  let codeblock_regex = Regex::new(r"^\s*[~`]{3}").unwrap();
+  if codeblock_regex.is_match(text) {
+    text.to_string()
+  } else {
+    format!("```\n{}\n```", text)
+  }
+}
+
+/// Replace JSDoc like links (`{@link http://example.com}`) with markdown links
+fn replace_links(text: &str) -> String {
+  let jsdoc_links_regex = Regex::new(r"(?i)\{@(link|linkplain|linkcode) (https?://[^ |}]+?)(?:[| ]([^{}\n]+?))?\}").unwrap();
+  jsdoc_links_regex
```

Comment:
Maybe in the future we could use  https://swc.rs/rustdoc/jsdoc/ for this purpose?

Reply:
🤷  I suspect it won't return things exactly the way vscode is interpreting here, but worth looking into... will add it to the list...